### ISSUE TITLE
False negative for S105 when variable is typed

### ISSUE THREAD

**Issue by zxdavb** [astral-sh/ruff#14991]
I note that `ruff check` detects the following:
```python
TEST_PASSWORD = "P@ssw0rd!!"  # Possible hardcoded password assigned to: "TEST_PASSWORD"
```

... except when the appropriate `noqa` directive is applied:
```python
TEST_PASSWORD = "P@ssw0rd!!"  # noqa: S105
```

However, these also pass a `ruff check` without complaint:
```python
TEST_PASSWORD: str = "P@ssw0rd!!"
TEST_PASSWORD: Final = "P@ssw0rd!!"
```

... and, IMO, they should not.

Ruff 0.8.3
Python 3.13.0
Ubuntu 24.04.1 LTS on WSL

**Comment by zxdavb** [comment]
A bit like this, I guess:

```rust
        Stmt::AnnAssign(
            assign_stmt @ ast::StmtAnnAssign {
                target,
                value,
                annotation,
                ..
            },
        ) => {
...

            if checker.enabled(Rule::HardcodedPasswordString) {
                flake8_bandit::rules::assign_hardcoded_password_string(checker, value, targets);
            }
        }
        Stmt::TypeAlias(ast::StmtTypeAlias { name, .. }) => {
```

**Comment by MichaReiser** [comment]
What you have seems almost correct and impressive that you found `std::slice::from_ref`. I remember that I only learned about that method after using Rust for several months:


```rust
            if checker.enabled(Rule::HardcodedPasswordString) {
                if let Some(value) = value.as_deref() {
                    flake8_bandit::rules::assign_hardcoded_password_string(
                        checker,
                        value,
                        std::slice::from_ref(target),
                    );
                }
            }
```
